src/main/java/com/redhat/jenkins/plugins/bayesian/Bayesian.java

```diff
@@ -137,6 +153,71 @@ public BayesianStepResponse submitStackForAnalysis(Collection<FilePath> manifest
             gson = null;
         }
     }
+    
+    public boolean isScanEnabled() throws BayesianException {
+        boolean flag = false;
+        String url = getOSIOCodebaseUrl();
+        Gson gson;
+        Codebase responseObj;
+        InputStream is = null;
+        BufferedReader br = null;
+
+        HttpGet httpGet = new HttpGet(url);
+        try (CloseableHttpClient client = HttpClients.createDefault();
+                CloseableHttpResponse response = client.execute(httpGet)) {
+
+                HttpEntity entity = response.getEntity();
+                is = entity.getContent();
+                StringBuilder sb = new StringBuilder();
+                String line;
+                br = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                gson = new GsonBuilder().create();
+                String jsonStr = sb.toString().replace("cve-scan", "cvescan");
+                responseObj = gson.fromJson(jsonStr, Codebase.class);
+
```

Comment:
I'd add refactor the whole code to read & parse JSON to its own utility method that return deserialized data or throw BayessianError

Reply:
refactored.